compiler/ghc/default.nix

```diff
@@ -323,6 +323,12 @@ in let configured-src = stdenv.mkDerivation (rec {
         -e 's/ghcprog="ghc-/ghcprog="${targetPrefix}ghc-/' \
         $i
     done
+
+    # Use absolute path to ar when cross-compile
```

Comment:
```suggestion
    # Use absolute path to ar when cross-compiling
```

Reply:
Why do we need to do this only for `ar`?

Reply:
Because only ar have no absolute path in settings (but gcc, ld, ranlib have). Just check it with ghc --info
So, if you have ghc built in pkgsCross.musl64 (all tools are prefixed) and use it to build ghc in pkgsMusl (unprefixed tools) you will get an error "x86_64-unknown-linux-musl-ar not found", because in native environment ar is just ar without system prefix.

Reply:
There is
```
export AR="${targetCC.bintools.bintools}/bin/${targetCC.bintools.targetPrefix}ar"
```
in the compiler/ghc/default.nix, so I understand that this is exported correctly, but ghc's configure doesn't pick it up correctly?

Reply:
Yes, configure script just takes exe name for ar without full path.
It's log tail, all tools with full paths except ar and objdump. 
```
----------------------------------------------------------------------
Configure completed successfully.

   Building GHC version  : 8.4.4
          Git commit id  : -n 39e711436400ecde05961224fa37759f2c2ff862

   Build platform        : x86_64-unknown-linux
   Host platform         : x86_64-unknown-linux
   Target platform       : x86_64-unknown-linux

   Bootstrapping using   : /nix/store/lwvpq5nqcams45fdavldc2cahg12qb3i-ghc-8.4.4/bin/ghc
      which is version   : 8.4.4

   Using (for bootstrapping) : /nix/store/1v0410l20p343l6cvpijz8bi4if2ysh6-gcc-wrapper-9.2.0/bin/cc
   Using gcc                 : /nix/store/04apk01xdv0j1jgbgj79fg0wip2m1plr-x86_64-unknown-linux-musl-stage-final-gcc-debug-wrapper-9.2.0/bin/x86_64-unknown-linux-musl-cc
      which is version       : 9.2.0
   Building a cross compiler : NO
   Unregisterised            : NO
   hs-cpp       : /nix/store/04apk01xdv0j1jgbgj79fg0wip2m1plr-x86_64-unknown-linux-musl-stage-final-gcc-debug-wrapper-9.2.0/bin/x86_64-unknown-linux-musl-cc
   hs-cpp-flags : -E -undef -traditional
   ar           : x86_64-unknown-linux-musl-ar
   ld           : /nix/store/qg8njpsk14yv0nyaxkvvi0q7ky98x69g-x86_64-unknown-linux-musl-binutils-wrapper-2.31.1/bin/x86_64-unknown-linux-musl-ld
   nm           : /nix/store/47hgsykqkslf8ybgs6g4899gghx5h5cl-x86_64-unknown-linux-musl-binutils-2.31.1/bin/x86_64-unknown-linux-musl-nm
   libtool      : 
   objdump      : x86_64-unknown-linux-musl-objdump
   ranlib       : /nix/store/47hgsykqkslf8ybgs6g4899gghx5h5cl-x86_64-unknown-linux-musl-binutils-2.31.1/bin/x86_64-unknown-linux-musl-ranlib
   windres      : 
   dllwrap      : 
   genlib       : 
   Happy        : /nix/store/0rcnjh7h986xxkvgip5c30vbp29mx3ry-happy-1.19.11-exe-happy/bin/happy (1.19.11)
   Alex         : /nix/store/spg0j23i2cfg1rslksmr61khjd6yfh02-alex-3.2.4-exe-alex/bin/alex (3.2.4)
   Perl         : /nix/store/yvmfc6lcsljxb8mdpaz63yl1j0z2zzm2-perl-5.30.1/bin/perl
   sphinx-build : /nix/store/q0bq2h0k2wqq8v42z17h9rclnigmax7z-python3.7-sphinx-1.7.9/bin/sphinx-build
   xelatex      : 

   Using LLVM tools
      clang : 
      llc   : 
      opt   : 
   HsColour : /nix/store/5blhmm84m8n5qbi2qlsgaakbf68as1r2-hscolour-1.24.4-exe-HsColour/bin/HsColour

   Tools to build Sphinx HTML documentation available: YES
   Tools to build Sphinx PDF documentation available: NO
----------------------------------------------------------------------

For a standard build of GHC (fully optimised with profiling), type (g)make.

To make changes to the default build configuration, copy the file
mk/build.mk.sample to mk/build.mk, and edit the settings in there.

For more information on how to configure your GHC build, see
   http://ghc.haskell.org/trac/ghc/wiki/Building

```

Reply:
That appears to be a bug in the `configure.ac` script then. We might want to fix this properly, through a patch in haskell.nix and an upstream MR instead of layering hacks ontop of things.

Reply:
Ohh this is for 8.4.4. We'll if we can verify this doesn't need to be done with recent GHCs, I'm happy with a rewrite hack (for 8.4.4).

Reply:
Same problem for 8.6.5 and 8.8.3. Probably for 8.10.1 too. I'll try to figure out how to fix it inside configure.ac, but my autotools-foo is very poor. :)

Reply:
It does some extra dance for `LD`, but here's for example `strip`:
https://github.com/ghc/ghc/blob/0bf640b19d7a7ad0800152752a71c1dd4e6c696d/configure.ac#L646-L650

So I guess we just need similar sections for AR and OBJDUMP

Reply:
There is logic for OBJDUMP though, not sure why it's not picked up...

https://github.com/ghc/ghc/blob/0bf640b19d7a7ad0800152752a71c1dd4e6c696d/configure.ac#L624-L632

Reply:
I see we don't set OBJDUMP in the ghc/default.nix, maybe we should?

Reply:
No, we need only ar here.
I fixed configure to respect $AR environment variable (patched aclocal.m4), now we have full path to ar in ghc settings.